File: Loon-0.5-Alpha/Loon-Neo/src/loon/utils/json/JsonSink.java
/**
 * Copyright 2008 - 2015 The Loon Game Engine Authors
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.5
 */
package loon.utils.json;

import loon.Json;
import loon.utils.ArrayMap;
import loon.utils.TArray;

public interface JsonSink<I extends JsonSink<I>> {

  I array(TArray<Object> c);

  I array(Json.Array c);

  I array(String key, TArray<Object> c);

  I array(String key, Json.Array c);

  I object(ArrayMap map);

  I object(Json.Object map);

  I object(String key, ArrayMap map);

  I object(String key, Json.Object map);

  I nul();

  I nul(String key);

  I value(Object o);

  I value(String key, Object o);

  I value(String s);

  I value(boolean b);

  I value(Number n);

  I value(String key, String s);

  I value(String key, boolean b);

  I value(String key, Number n);

  I array();

  I object();

  I array(String key);

  I object(String key);

  I end();
}

File: C#/Loon2MonoGame/LoonMonoGame-Lib/loon/utils/json/JsonSink.cs
using java.lang;

namespace loon.utils.json
{

	public interface JsonSink<I> where I : JsonSink<I>
	{

		I Array(TArray<object> c);

		I Array(loon.Json_Array c);

		I Array(string key, TArray<object> c);

		I Array(string key, loon.Json_Array c);

		I Object(ArrayMap map);

		I Object(loon.Json_Object map);

		I Object(string key, ArrayMap map);

		I Object(string key, loon.Json_Object map);

		I Nul();

		I Nul(string key);

		I Value(object o);

		I Value(string key, object o);

		I Value(string s);

		I Value(bool b);

		I Value(double n);

		I Value(Number n);

		I Value(string key, string s);

		I Value(string key, bool b);

		I Value(string key, double n);

		I Value(string key, Number n);

		I Array();

		I Object();

		I Array(string key);

		I Object(string key);

		I End();
	}

}
